File: src/main/java/gruvexp/bbminigames/menu/PlayerMenuRow.java
package gruvexp.bbminigames.menu;

import gruvexp.bbminigames.Main;
import gruvexp.bbminigames.twtClassic.BotBowsPlayer;
import org.bukkit.Material;
import org.bukkit.NamespacedKey;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.inventory.meta.SkullMeta;
import org.bukkit.persistence.PersistentDataType;

import java.util.function.Consumer;

public class PlayerMenuRow extends MenuRow{

    public PlayerMenuRow(Inventory inventory, String menuActionId, int startSlot, int size) {
        super(inventory, menuActionId, startSlot, size);
    }

    @Override
    public void addItem(ItemStack item) {
        if (item.getType() != Material.PLAYER_HEAD) throw new IllegalArgumentException("The item must be a player head");
        super.addItem(item);
    }

    public ItemStack getItem(BotBowsPlayer bp) {
        for (ItemStack item : itemList) {
            SkullMeta meta = (SkullMeta) item.getItemMeta();
            NamespacedKey key = new NamespacedKey(Main.getPlugin(), "uuid");
            String storedUUID = meta.getPersistentDataContainer().get(key, PersistentDataType.STRING);
            if (storedUUID.equals(bp.avatar.getUUID().toString())) {
                return item;
            }
        }
        return null;
    }

    public void editItem(BotBowsPlayer bp, Consumer<ItemMeta> action) {
        ItemStack item = getItem(bp);
        if (item == null) return;
        item.editMeta(action);
        if (isVisible) displayRow();
    }

    public void removeItem(BotBowsPlayer bp) {
        removeItem(getItem(bp));
    }
}

File: src/main/java/gruvexp/bbminigames/menu/PlayerMenuRow.kt
package gruvexp.bbminigames.menu

import gruvexp.bbminigames.Main
import gruvexp.bbminigames.twtClassic.BotBowsPlayer
import org.bukkit.Material
import org.bukkit.NamespacedKey
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.ItemMeta
import org.bukkit.persistence.PersistentDataType

class PlayerMenuRow(inventory: Inventory, menuActionId: String, startSlot: Int, size: Int) :
    MenuRow(inventory, menuActionId, startSlot, size) {
    override fun addItem(item: ItemStack) {
        require(item.type == Material.PLAYER_HEAD) { "The item must be a player head" }
        super.addItem(item)
    }

    fun getItem(bp: BotBowsPlayer): ItemStack? {
        items.filterNotNull().forEach {
            val key = NamespacedKey(Main.getPlugin(), "uuid")
            val storedUUID = it.itemMeta.persistentDataContainer.get(key, PersistentDataType.STRING)
            if (storedUUID == bp.avatar.uuid.toString()) {
                return@getItem it
            }
        }
        return null
    }

    fun editItem(bp: BotBowsPlayer, action: ItemMeta.() -> Unit) {
        val item = getItem(bp) ?: return
        item.editMeta(action)
        if (isVisible) displayRow()
    }

    fun removeItem(bp: BotBowsPlayer) {
        removeItem(getItem(bp))
    }
}
